python/ray/tune/tests/test_trial_runner_3.py

```diff
@@ -510,8 +510,8 @@ def testTrialSaveRestore(self):
         runner2.step()  # Process save
         self.assertRaises(TuneError, runner2.step)
 
-    def testTrialNoSave(self):
-        """Check that non-checkpointing trials are not saved."""
+    def testTrialNoCheckpointSave(self):
+        """Check that non-checkpointing trials *are* saved."""
```

Comment:
@richardliaw quick question: In this PR, we change the behavior that even non-checkpointing trials *are* saved. The reason is that we often got out-of-sync checkpoints (the FS sync already synchronized new checkpoints to the driver, but the trial metadata has not been updated on the driver side, yet). 
Also, I don't see why we wouldn't want to save intermediate state of non-checkpointing trials. It seems to me like the main reason here would be that we can't restore these trials. However, in the case of early-exiting an experiment via keyboard interrupt, we sometimes may want to analyze reported resuls until that time.

Thus, I quickly wanted to confirm if there is any other reason why we specifically want trials not to be saved when they are not checkpointing.

Reply:
What does it mean to save an intermediate trial without checkpoints? Do you mean just to move the trial folder back to local host?

Reply:
"Saving" in that sense is to store its metadata (e.g. last results) in the experiment checkpoint (trial runner checkpoint)